react/src/stores/priorityFee/useSyncPriorityFeeStore.ts

```diff
@@ -0,0 +1,210 @@
+import {
+	PriorityFeeCalculator,
+	SANITY_CHECK_ABS_MAX_FEE_IN_SOL,
+} from '@drift/common';
+import { useCallback, useEffect, useRef } from 'react';
+import { useImmediateInterval } from '../../hooks/useImmediateInterval';
+import { usePriorityFeesPollingRate } from '../../hooks/priorityFees/usePriorityFeesPollingRate';
+
+const RECENT_SAMPLES: number[][] = []; // array of slots, each slot is an array of priority fees
+let LAST_SEEN_SLOT_IN_SAMPLES = 0;
+
+/**
+ * Custom priority fee strategy that calculates the priority fee based on the recent samples.
+ * New samples are provided to the calculate function, and the strategy uses the given percentile of
+ * the the last 10 batches of recent samples as a benchmark for the priority fee to be used.
+ */
+const createPercentilePriorityFeeStrategy = (
+	targetPercentile: number
+): PriorityFeeStrategy => {
+	const percentilePriorityFeeStrategy: PriorityFeeStrategy = {
+		calculate(newSamples: SolanaPriorityFeeResponse[]) {
+			if (!newSamples?.length) return 0;
+
+			const filteredSamples = newSamples.filter(
+				(sample) => sample.slot > LAST_SEEN_SLOT_IN_SAMPLES
+			);
+
+			RECENT_SAMPLES.unshift(newSamples.map((s) => s.prioritizationFee));
+			RECENT_SAMPLES.splice(FEE_SUBSCRIPTION_SLOT_LOOKBACK);
+
+			const allRecentSamplesAscendingSorted = RECENT_SAMPLES.flat().sort(
+				(a, b) => {
+					return a - b;
+				}
+			);
+
+			const targetPercentileIndex = Math.min(
+				allRecentSamplesAscendingSorted.length - 1,
+				Math.ceil(
+					(allRecentSamplesAscendingSorted.length / 100) * targetPercentile
+				)
+			);
+
+			const shouldSplitWithBelow =
+				targetPercentile < 100 &&
+				targetPercentileIndex >= allRecentSamplesAscendingSorted.length - 1; // If the number of samples being returned are sufficiently small then in practise the target percentile is just selecting the MAX priority fee every time. For some safety, average top two instead when we get this case.
+
+			const pFee = shouldSplitWithBelow
+				? (allRecentSamplesAscendingSorted[targetPercentileIndex] +
+						allRecentSamplesAscendingSorted[targetPercentileIndex - 1]) /
+				  2
+				: allRecentSamplesAscendingSorted[targetPercentileIndex];
+
+			LAST_SEEN_SLOT_IN_SAMPLES = Math.max(
+				...filteredSamples.map((sample) => sample.slot)
+			);
+
+			return pFee;
+		},
+	};
+
+	return percentilePriorityFeeStrategy;
+};
+
+export const useSyncPriorityFeeStore = ({
+	heliusRpcUrl,
+	feeSubscriptionAddressStrings = PRIORITY_FEE_SUBSCRIPTION_ADDRESS_STRINGS,
+	targetFeePercentile,
+	userPriorityFeeType,
+	userCustomMaxPriorityFeeCap,
+	userCustomPriorityFee,
+}: {
+	heliusRpcUrl: string;
+	feeSubscriptionAddressStrings?: string[];
+	targetFeePercentile: number;
+	userPriorityFeeType: FeeType;
+	userCustomMaxPriorityFeeCap: number;
+	userCustomPriorityFee: number | null;
+}) => {
+	const percentilePriorityFeeStrategy =
+		createPercentilePriorityFeeStrategy(targetFeePercentile);
```

Comment:
I think can replace this with `PriorityFeeStrategyFactory.movingWindowTargetPercentileStrategy`

Reply:
guessing I was halfway through this when you took over, I remember creating that seperate file to hold the different strategies